src/frontends/lean/elaborator.cpp

```diff
@@ -3405,6 +3405,7 @@ expr elaborator::visit_pi(expr const & e) {
 expr elaborator::visit_let(expr const & e, optional<expr> const & expected_type) {
     expr ref = e;
     expr new_type  = visit(let_type(e), none_expr());
+    new_type = ensure_type(new_type, let_type(e));
```

Comment:
Every other call to `ensure_type` in this file seems to call `get_ref_for_child` in order to obtain the second argument to this call:
https://github.com/leanprover-community/lean/blob/508db1acd2249ac7bc0654867ab73a6d5c0a6408/src/frontends/lean/elaborator.cpp#L3387-L3389

Is there a reason that doesn't make sense here?

Reply:
I guess it's a reasonable precaution.